### PR TITLE
test(auth): add gen2 auth environments

### PR DESCRIPTION
**Pull request by zchenwei**
#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [x] Relevant documentation is changed or added (and PR referenced)
- [x] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/e2e/features/ui/components/authenticator/gen2/sign-up-with-phone.feature [hunk 1]
```diff
@@ -0,0 +1,36 @@
+Feature: Sign Up with Phone 
+
+  Create a new user in the Amazon Cognito UserPool by passing the new user’s phone number and password.
+
+  Background:
+    Given I'm running the example "ui/components/authenticator/gen2/sign-up-with-phone/"
+    Then I intercept '{ "headers": { "X-Amz-Target": "AWSCognitoIdentityProviderService.SignUp" } }' with fixture "sign-up-with-phone"
+
+  @angular @react @vue @react-native
```
Comment (reesscot) [inline packages/e2e/features/ui/components/authenticator/gen2/sign-up-with-phone.feature:9]: Do you know if cucumber has a way to share fragments of feature files? Might be something to look into so we don't have to copy the entire files.

File: environments/auth/auth-with-email-lambda-signup-trigger/package.json [hunk 2]
```diff
@@ -3,6 +3,6 @@
   "name": "@aws-amplify/ui-auth-with-email-lambda-signup-trigger-environment",
   "version": "0.0.1",
   "scripts": {
-    "pull": "amplify pull --appId d1ytpi9jtd8llt --envName staging"
```
Comment (reesscot) [inline environments/auth/auth-with-email-lambda-signup-trigger/package.json:6]: I think we'll want to continue testing with Gen1 as well, right? I may be missing something, but it looks like this is removing that capability

Comment (zchenwei) [reply environments/auth/auth-with-email-lambda-signup-trigger/package.json:6]: I combine both to reuse the shell script. It was moved to the end

Comment (zchenwei) [reply environments/auth/auth-with-email-lambda-signup-trigger/package.json:6]: `xxx && echo n | amplify pull --appId d1ytpi9jtd8llt --envName staging"`

File: packages/e2e/features/ui/components/authenticator/gen2/sign-up-with-phone.feature (hunk 1 above)
Comment (zchenwei) [reply packages/e2e/features/ui/components/authenticator/gen2/sign-up-with-phone.feature:9]: I can check but likely no way. Even though we can change the route, I am afraid we would have to copy the same scenarios for gen2 in the file

**Pull request closed without merging**